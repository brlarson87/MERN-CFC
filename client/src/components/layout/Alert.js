//CORE REACT
import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

const Alert = ({ alert }) =>
  alert && <div className={`alert alert--${alert.alertType}`}>{alert.msg}</div>;

Alert.propTypes = {
  alerts: PropTypes.object,
};

const mapStateToProps = (state) => ({
  alert: state.alert,
});

export default connect(mapStateToProps)(Alert);
