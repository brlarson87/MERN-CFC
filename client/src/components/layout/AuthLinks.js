//CORE REACT
import React, { Fragment, useEffect } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

//Actions
import { logout } from "../../actions/auth";

//Utils
import ticketStatus from "../../utils/ticketStatus";

const AuthLinks = ({ user, loading, logout }) => {
  useEffect(() => {
    let menuBtn = document.querySelector(".menu-btn");
    let ticket = document.querySelector(".header__icon-ticket");
    let popOut = document.querySelector(".pop-out-left");
    let navs = document.querySelector(".main-nav-menu");
    let overlay = document.querySelector(".overlay-all");

    let navLinks = document.querySelectorAll(".main-nav-menu__item__link");

    const close = () => {
      menuBtn.classList.remove("open");
      overlay.classList.remove("show-overlay");
      popOut.classList.remove("show");
      ticket.classList.remove("move-ticket");
      navs.classList.remove("nav-showing");
    };

    for (let i = 0; i < navLinks.length; i++) {
      navLinks[i].addEventListener("click", close);
    }

    menuBtn.addEventListener("click", () => {
      menuBtn.classList.toggle("open");
      overlay.classList.toggle("show-overlay");
      popOut.classList.toggle("show");
      ticket.classList.toggle("move-ticket");
      navs.classList.toggle("nav-showing");
    });

    overlay.addEventListener("click", close);
  }, []);

  return (
    <Fragment>
      <div className='overlay-all'></div>
      <label htmlFor='navi-toggle' className='menu-btn'>
        <span className='menu-icon'>&larr;</span>
      </label>
      <div className='header__icon-ticket'>
        <i className='fas fa-ticket-alt'>
          &nbsp; x {!loading && user && ticketStatus(user.tickets).active}
        </i>
      </div>

      <div className='pop-out-left'>
        <ul className='main-nav-menu'>
          <li className='main-nav-menu__item'>
            <Link
              to='/about'
              className='main-nav-menu__item__link'
              onClick={() => window.scrollTo(0, 0)}
            >
              <i className='fas fa-hand-holding-medical'></i> &nbsp; About Us
            </Link>
          </li>
          <li className='main-nav-menu__item'>
            <Link
              to='/cars'
              className='main-nav-menu__item__link'
              onClick={() => window.scrollTo(0, 0)}
            >
              <i className='fas fa-car'></i>&nbsp; Cars
            </Link>
          </li>
          <li className='main-nav-menu__item'>
            <Link
              to='/charities'
              className='main-nav-menu__item__link'
              onClick={() => window.scrollTo(0, 0)}
            >
              <i className='fas fa-ribbon'></i>&nbsp; Charities
            </Link>
          </li>
          <li className='main-nav-menu__item'>
            <Link
              to='/account'
              className='main-nav-menu__item__link'
              onClick={() => window.scrollTo(0, 0)}
            >
              <i className='far fa-user'>
                &nbsp; {!loading && user && user.firstName}
              </i>
            </Link>
          </li>
          <li className='main-nav-menu__item'>
            <Link to='/' className='main-nav-menu__item__link' onClick={logout}>
              <i className='fas fa-sign-out-alt'></i>&nbsp; Logout
            </Link>
          </li>
          {user && user.admin && 
          <li className='main-nav-menu__item'>
            <Link to='/admin-priv' className='main-nav-menu__item__link' onClick={logout}>
              <i className='fas fa-sign-out-alt'></i>&nbsp; Admin
            </Link>
          </li>}
          
        </ul>
      </div>
    </Fragment>
  );
};

AuthLinks.propTypes = {
  user: PropTypes.object,
  loading: PropTypes.bool,
};

const mapStateToProps = (state) => ({
  user: state.auth.user,
  loading: state.auth.loading,
});

export default connect(mapStateToProps, { logout })(AuthLinks);
