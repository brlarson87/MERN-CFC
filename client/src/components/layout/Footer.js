//CORE REACT
import React from "react";
import { Link } from "react-router-dom";

const Footer = () => {
  return (
    <div className='footer'>
      <h4 className='footer__social-header'>Follow us</h4>

      <div className='footer__container'>
        <ul className='footer__nav footer__nav--left'>
          <li className='footer__nav--item'>
            <Link to='/' className='footer__nav--link'>
              <i className='fab fa-twitter-square'></i>&nbsp; Twitter
            </Link>
          </li>
          <li className='footer__nav--item'>
            <Link to='/' className='footer__nav--link'>
              <i className='fab fa-facebook-square'></i>&nbsp; Facebook
            </Link>
          </li>
          <li className='footer__nav--item'>
            <Link to='/' className='footer__nav--link'>
              <i className='fab fa-youtube'></i>&nbsp; Youtube
            </Link>
          </li>
          <li className='footer__nav--item'>
            <Link to='/' className='footer__nav--link'>
              <i className='fab fa-instagram'></i>&nbsp; Instagram
            </Link>
          </li>
        </ul>

        <ul className='footer__nav footer__nav--right'>
          <li className='footer__nav--item footer__nav--item--sub'>
            <Link to='/' className='footer__nav--link'>
              Support the site
            </Link>
          </li>
          <li className='footer__nav--item footer__nav--item--sub'>
            <Link to='/' className='footer__nav--link'>
              Contact Us
            </Link>
          </li>
          <li className='footer__nav--item footer__nav--item--sub'>
            <Link to='/' className='footer__nav--link'>
              Submit a charity
            </Link>
          </li>
          <li className='footer__nav--item footer__nav--item--sub'>
            <Link to='/' className='footer__nav--link'>
              Archived cars
            </Link>
          </li>
          <li className='footer__nav--item footer__nav--item--sub'>
            <Link to='/' className='footer__nav--link'>
              Archived charity winners
            </Link>
          </li>
        </ul>
      </div>
      <h5 className='footer__copyright-tag'>&copy; 2019 Cars for Causes</h5>
    </div>
  );
};

export default Footer;
