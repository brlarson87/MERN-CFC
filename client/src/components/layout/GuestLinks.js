//CORE REACT
import React, { Fragment, useEffect } from "react";
import { Link } from "react-router-dom";

const GuestLinks = () => {
  useEffect(() => {
    let menuBtn = document.querySelector(".menu-btn");
    let popOut = document.querySelector(".pop-out-left");
    let navs = document.querySelector(".main-nav__menu");
    let overlay = document.querySelector(".overlay-all");

    let navLinks = document.querySelectorAll(".main-nav__menu--item-link");

    const close = () => {
      menuBtn.classList.remove("open");
      overlay.classList.remove("show-overlay");
      popOut.classList.remove("show");
      navs.classList.remove("nav-showing");
    };

    for (let i = 0; i < navLinks.length; i++) {
      navLinks[i].addEventListener("click", close);
    }

    menuBtn.addEventListener("click", () => {
      menuBtn.classList.toggle("open");
      overlay.classList.toggle("show-overlay");
      popOut.classList.toggle("show");
      navs.classList.toggle("nav-showing");
    });

    overlay.addEventListener("click", close);
  }, []);

  return (
    <Fragment>
      <div className='overlay-all'></div>
      <label htmlFor='navi-toggle' className='menu-btn'>
        <span className='menu-icon'>&larr;</span>
      </label>
      <div className='pop-out-left'>
        <ul className='main-nav__menu'>
          <li className='main-nav__menu--item'>
            <Link
              to='/about'
              className='main-nav__menu--item-link menu-link-active'
            >
              <i className='fas fa-hand-holding-medical'></i> &nbsp; About Us
            </Link>
          </li>
          <li className='main-nav__menu--item'>
            <Link to='/cars' className='main-nav__menu--item-link'>
              <i className='fas fa-car'></i>&nbsp; Cars
            </Link>
          </li>
          <li className='main-nav__menu--item'>
            <Link to='/charities' className='main-nav__menu--item-link'>
              <i className='fas fa-ribbon'></i>&nbsp; Charities
            </Link>
          </li>
          <li className='main-nav__menu--item'>
            <Link to='/login' className='main-nav__menu--item-link'>
              <i className='fas fa-sign-in-alt'></i>&nbsp; Login
            </Link>
          </li>
          <li className='main-nav__menu--item'>
            <Link to='/register' className='main-nav__menu--item-link'>
              <i className='fas fa-registered'></i>&nbsp; Register
            </Link>
          </li>
        </ul>
      </div>
    </Fragment>
  );
};

export default GuestLinks;
