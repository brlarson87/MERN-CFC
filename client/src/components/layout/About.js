import React, { Fragment } from "react";

import Footer from "../layout/Footer";

const Layout = () => {
  return (
    <Fragment>
      <div class='container'>
        <div class='about-us'>
          <div class='about-us__text-box'>
            <h2 class='about-us__text-box--title'>About Us</h2>
            <p class='about-us__text-box--text'>
              Lorem, ipsum dolor sit amet consectetur adipisicing elit. Eos, quo
              id unde tenetur reprehenderit consequatur ut deleniti. Nobis
              nostrum, sequi delectus voluptatum itaque iure explicabo.
              Quibusdam blanditiis consequatur quis vitae expedita dolorum
              cupiditate, eius dolores voluptate. Blanditiis labore quidem et
              nisi corporis quod doloribus assumenda explicabo fugit iure.
              Corporis ipsum porro, laboriosam perspiciatis nulla, reprehenderit
              praesentium quaerat assumenda fuga accusamus atque sint quia
              repellendus velit nostrum. Voluptatibus fugiat fugit, doloribus
              unde est debitis provident ullam porro incidunt adipisci iure
              necessitatibus repudiandae, expedita nam distinctio aliquid
              cupiditate! Eum cumque, libero voluptatem provident accusantium
              nihil, rerum maxime vel aliquid quam vero molestiae.
            </p>
          </div>
        </div>
      </div>
      <Footer />
    </Fragment>
  );
};

export default Layout;
