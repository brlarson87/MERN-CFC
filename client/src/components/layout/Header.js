import React, { Fragment } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";

// Header nav components
import GuestLinks from "./GuestLinks";
import AuthLinks from "./AuthLinks";

const Header = ({ loading, isAuthenticated }) => {
  return (
    <Fragment>
      <header className='header'>
        <nav className='main-nav'>
          <div className='main-nav__logo'>
            <Link to='/' style={{ textDecoration: "none" }}>
              <h2 className='main-nav__logo-title'>CarsforCauses</h2>
            </Link>
          </div>
        </nav>
        {/*<input type='checkbox' class='menu-btn__checkbox' id='navi-toggle' />*/}

        {!loading && isAuthenticated ? <AuthLinks /> : <GuestLinks />}
      </header>
    </Fragment>
  );
};

Header.propTypes = {
  loading: PropTypes.bool,
  isAuthenticated: PropTypes.bool
};

const mapStateToProps = state => ({
  loading: state.auth.loading,
  isAuthenticated: state.auth.isAuthenticated
});

export default connect(mapStateToProps)(Header);
