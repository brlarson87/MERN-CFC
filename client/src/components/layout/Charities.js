//CORE REACT
import React, { Fragment, useEffect, useState } from "react";
import { connect } from "react-redux";

//Components
import Header from "./Header";
import TableRow from "./TableRow";
import CharityConfirm from "../modals/CharityConfirm";
//import Spinner from "./Spinner";

//Actions
import { loadCharities } from "../../actions/charities";
import { loadPrizes } from "../../actions/prizes";
import { showCharityConfirmation } from "../../actions/modal";

//Utils
import { utilFilter } from "../../utils/utilFilter";
import { totalCountOfCharities } from "../../utils/utilFilter";

const Charities = ({
  loadCharities,
  loadPrizes,
  showCharityConfirmation,
  charities,
  loading,
  user,
  prizes,
  modalShow,
}) => {
  useEffect(() => {
    loadCharities();
    loadPrizes();
    const charityBtn = document.querySelector(".charityBtn");
    charityBtn.addEventListener = document.addEventListener(
      "click",
      () => (charityBtn.style.opacity = "0")
    );
    window.addEventListener("scroll", () => {
      const {
        scrollTop,
        scrollHeight,
        clientHeight,
      } = document.documentElement;

      if (scrollTop + clientHeight >= scrollHeight - 5) {
        charityBtn.style.opacity = "1";
      } else {
        charityBtn.style.opacity = "0";
      }
    });
  }, [loadCharities, loadPrizes]);

  const [charityData, setCharityData] = useState({
    count: 10,
    filter: "",
  });

  const { count, filter } = charityData;

  const onChange = (e) =>
    setCharityData({ ...charityData, [e.target.name]: e.target.value });

  const incrementCharityCount = () => {
    setTimeout(() => {
      setCharityData({ ...charityData, count: count + 10 });
    }, 500);
  };

  return (
    <Fragment>
      <Header />
      {/*****************************************/}
      {/*----------CHARITYCONFIRM COMPONENT----------*/}
      {/*****************************************/}
      {modalShow && <CharityConfirm />}
      <div className='container'>
        <div className='search-form'>
          <input
            type='text'
            name='filter'
            value={filter}
            onChange={(e) => onChange(e)}
            className='search-form__input'
            placeholder='Search charities...'
            autoComplete='off'
          />
          <div className='search-form__submit-btn'>
            <i className='fas fa-search'></i>
          </div>
        </div>
        <table className='charity-table'>
          <thead className='charity-table__head'>
            <tr className='charity-table__row'>
              <th className='charity-table__head'>Name</th>
              <th className='charity-table__head charity-table__head--lg'>
                Description
              </th>
              <th className='charity-table__head charity-table__head--sm'>
                Website
              </th>
              <th className='charity-table__head charity-table__head--sm'>
                Enter this Charity
              </th>
            </tr>
          </thead>

          <tbody className='charity-table__body'>
            {/*****************************************/}
            {/*----------TABLEROW COMPONENT-----------*/}
            {/*****************************************/}
            {charities &&
              !loading &&
              utilFilter(charities, filter)
                .slice(0, count)
                .map((charity, i) => (
                  <TableRow
                    key={i}
                    user={user}
                    prizes={prizes}
                    showCharityConfirmation={showCharityConfirmation}
                    charity={charity}
                  />
                ))}
          </tbody>
        </table>
        {/* {loading && <Spinner />} */}
        {charities && !totalCountOfCharities(charities, filter) && !loading && (
          <div className='match-icon-box u-margin-top-lg'>
            <h1>No matches...</h1>
            <i className='fas fa-sad-tear u-lg-font u-margin-top-lg'></i>
          </div>
        )}

        <button
          className='charityBtn'
          style={{ opacity: 0 }}
          onClick={incrementCharityCount}
        >
          Load more
        </button>
      </div>
    </Fragment>
  );
};

const mapStateToProps = (state) => ({
  charities: state.charities.charities,
  loading: state.charities.loading,
  user: state.auth.user,
  prizes: state.prizes.prizes,
  modalShow: state.modal.show,
});

export default connect(mapStateToProps, {
  loadCharities,
  loadPrizes,
  showCharityConfirmation,
})(Charities);
