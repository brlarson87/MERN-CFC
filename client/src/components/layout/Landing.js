//CORE REACT
import React, { Fragment, useEffect } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import Header from "./Header";

//import Spinner from "./Spinner";
import Footer from "./Footer";

const Landing = ({ loading, isAuthenticated, user }) => {
  useEffect(() => {
    let scroll =
      window.requestAnimationFrame ||
      function (callback) {
        window.setTimeout(callback, 1000 / 60);
      };

    let elementToShow = document.querySelector(".show-on-scroll");
    function loop() {
      if (isElementInViewport(elementToShow)) {
        elementToShow.classList.add("is-visible");
      } else {
        elementToShow.classList.remove("is-visible");
      }

      scroll(loop);
    }

    loop();

    function isElementInViewport(el) {
      let rect = el.getBoundingClientRect();
      return (
        (rect.top <= 0 && rect.bottom >= 0) ||
        (rect.bottom >=
          (window.innerHeight || document.documentElement.clientHeight) &&
          rect.top <=
            (window.innerHeight || document.documentElement.clientHeight)) ||
        (rect.top >= 0 &&
          rect.bottom <=
            (window.innerHeight || document.documentElement.clientHeight))
      );
    }
  }, []);

  return (
    <Fragment>
      <Header />
      <div className='car-bg'>
        <div className='see-through--top'>&nbsp;</div>
        <div className='landing-info'>
          <h2 className='landing-info--title'>
            Using the love of cars to crowdfund donations
          </h2>
          <div className='landing-info--grid'>
            {!loading && isAuthenticated && user ? (
              <Fragment>
                <div className='landing-info--box'>
                  <h4 className='landing-info--box--heading'>
                    Welcome Back {user.firstName}!
                  </h4>
                  <p className='landing-info--box--text margin-bottom-lg'>
                    Good luck in the pools.
                  </p>
                  <Link
                    to='/account'
                    className='btn-abs-bottom'
                    onClick={() => window.scrollTo(0, 0)}
                  >
                    My Account
                  </Link>
                </div>
                <div className='arrow-main'>&rarr;</div>
                <div className='landing-info--box'>
                  <h4 className='landing-info--box--heading'>
                    Purchase Tickets
                  </h4>
                  <p className='landing-info--box--text'>
                    Tickets are $1 each.
                  </p>
                  <Link to='/purchaseTickets' className='btn-abs-bottom'>
                    Buy Tickets
                  </Link>
                </div>
              </Fragment>
            ) : (
              <Fragment>
                <div className='landing-info--box'>
                  <h4 className='landing-info--box--heading'>
                    Create an account
                  </h4>
                  <p className='landing-info--box--text margin-bottom-lg'>
                    Sign up, It's free!
                  </p>
                  <Link to='/login' className='btn-abs-bottom'>
                    Signup/Login
                  </Link>
                </div>
                <div className='arrow-main'>&rarr;</div>
                <div className='landing-info--box'>
                  <h4 className='landing-info--box--heading'>
                    Purchase Tickets
                  </h4>
                  <p className='landing-info--box--text'>
                    Tickets are $1 each.
                  </p>
                </div>
              </Fragment>
            )}

            <div className='arrow-main'>&rarr;</div>
            <div className='landing-info--box'>
              <h4 className='landing-info--box--heading'>Enter tickets</h4>
              <p className='landing-info--box--text'>
                Browse the collection of car pools and find your dream car!
                Enter enough tickets, and you can also enter a charity to the
                corresponding charity pool.
              </p>
              <Link
                to='/cars'
                className='btn-abs-bottom'
                onClick={() => window.scrollTo(0, 0)}
              >
                Check out the cars
              </Link>
            </div>
          </div>
        </div>

        <div className='see-through--bottom'>
          <div className='see-through--bottom__box show-on-scroll'>
            <h1 className='see-through--bottom__title'>
              Browse the many charities you can enter.. or sumbit your own!
            </h1>
            <Link
              to='/charities'
              className='btn--link-2'
              onClick={() => window.scrollTo(0, 0)}
            >
              &rarr;
            </Link>
          </div>
        </div>
      </div>
      <Footer />
    </Fragment>
  );
};

Landing.propTypes = {
  user: PropTypes.object,
  isAuthenticated: PropTypes.bool,
  loading: PropTypes.bool,
};

const mapStateToProps = (state) => ({
  isAuthenticated: state.auth.isAuthenticated,
  loading: state.auth.loading,
  user: state.auth.user,
});

export default connect(mapStateToProps)(Landing);
