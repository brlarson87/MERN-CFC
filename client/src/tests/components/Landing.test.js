import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { Landing } from "../../components/layout/Landing";

import users from "../fixtures/users";

it("renders Landing page on inital load", () => {
  const wrapper = shallow(<Landing />);

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("renders Landing page with user authenticated", () => {
  const wrapper = shallow(
    <Landing user={users[0]} isAuthenticated={true} loading={false} />
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});
