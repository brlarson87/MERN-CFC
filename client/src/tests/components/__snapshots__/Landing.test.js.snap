// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Landing page on inital load 1`] = `
<Fragment>
  <Connect(Header) />
  <div
    className="car-bg"
  >
    <div
      className="see-through--top"
    >
       
    </div>
    <div
      className="landing-info"
    >
      <h2
        className="landing-info--title"
      >
        Using the love of cars to crowdfund donations
      </h2>
      <div
        className="landing-info--grid"
      >
        <div
          className="landing-info--box"
        >
          <h4
            className="landing-info--box--heading"
          >
            Create an account
          </h4>
          <p
            className="landing-info--box--text margin-bottom-lg"
          >
            Sign up, It's free!
          </p>
          <Link
            className="btn-abs-bottom"
            to="/login"
          >
            Signup/Login
          </Link>
        </div>
        <div
          className="arrow-main"
        >
          →
        </div>
        <div
          className="landing-info--box"
        >
          <h4
            className="landing-info--box--heading"
          >
            Purchase Tickets
          </h4>
          <p
            className="landing-info--box--text"
          >
            Tickets are $1 each.
          </p>
        </div>
        <div
          className="arrow-main"
        >
          →
        </div>
        <div
          className="landing-info--box"
        >
          <h4
            className="landing-info--box--heading"
          >
            Enter tickets
          </h4>
          <p
            className="landing-info--box--text"
          >
            Browse the collection of car pools and find your dream car! Enter enough tickets, and you can also enter a charity to the corresponding charity pool.
          </p>
          <Link
            className="btn-abs-bottom"
            onClick={[Function]}
            to="/cars"
          >
            Check out the cars
          </Link>
        </div>
      </div>
    </div>
    <div
      className="see-through--bottom"
    >
      <div
        className="see-through--bottom__box show-on-scroll"
      >
        <h1
          className="see-through--bottom__title"
        >
          Browse the many charities you can enter.. or sumbit your own!
        </h1>
        <Link
          className="btn--link-2"
          onClick={[Function]}
          to="/charities"
        >
          →
        </Link>
      </div>
    </div>
  </div>
  <Footer />
</Fragment>
`;

exports[`renders Landing page with user authenticated 1`] = `
<Fragment>
  <Connect(Header) />
  <div
    className="car-bg"
  >
    <div
      className="see-through--top"
    >
       
    </div>
    <div
      className="landing-info"
    >
      <h2
        className="landing-info--title"
      >
        Using the love of cars to crowdfund donations
      </h2>
      <div
        className="landing-info--grid"
      >
        <div
          className="landing-info--box"
        >
          <h4
            className="landing-info--box--heading"
          >
            Welcome Back 
            Blake
            !
          </h4>
          <p
            className="landing-info--box--text margin-bottom-lg"
          >
            Good luck in the pools.
          </p>
          <Link
            className="btn-abs-bottom"
            onClick={[Function]}
            to="/account"
          >
            My Account
          </Link>
        </div>
        <div
          className="arrow-main"
        >
          →
        </div>
        <div
          className="landing-info--box"
        >
          <h4
            className="landing-info--box--heading"
          >
            Purchase Tickets
          </h4>
          <p
            className="landing-info--box--text"
          >
            Tickets are $1 each.
          </p>
          <Link
            className="btn-abs-bottom"
            to="/purchaseTickets"
          >
            Buy Tickets
          </Link>
        </div>
        <div
          className="arrow-main"
        >
          →
        </div>
        <div
          className="landing-info--box"
        >
          <h4
            className="landing-info--box--heading"
          >
            Enter tickets
          </h4>
          <p
            className="landing-info--box--text"
          >
            Browse the collection of car pools and find your dream car! Enter enough tickets, and you can also enter a charity to the corresponding charity pool.
          </p>
          <Link
            className="btn-abs-bottom"
            onClick={[Function]}
            to="/cars"
          >
            Check out the cars
          </Link>
        </div>
      </div>
    </div>
    <div
      className="see-through--bottom"
    >
      <div
        className="see-through--bottom__box show-on-scroll"
      >
        <h1
          className="see-through--bottom__title"
        >
          Browse the many charities you can enter.. or sumbit your own!
        </h1>
        <Link
          className="btn--link-2"
          onClick={[Function]}
          to="/charities"
        >
          →
        </Link>
      </div>
    </div>
  </div>
  <Footer />
</Fragment>
`;
